packages/eas-cli/src/build/ios/bundleIdentifer.ts

```diff
@@ -1,76 +1,121 @@
         ],
       });
-      return bundleIdentifier;
+      switch (bundleIdentifierSource) {
+        case BundleIdenitiferSource.AppJson: {
+          IOSConfig.BundleIdenitifer.setBundleIdentifierForPbxproj(
+            projectDir,
+            bundleIdentifierFromConfig
+          );
+          break;
+        }
+        case BundleIdenitiferSource.XcodeProject: {
+          await updateAppJsonConfigAsync(projectDir, exp, bundleIdentifierFromPbxproj);
+          break;
+        }
+      }
     }
-  } else if (bundleIdentifierFromPbxproj === null && bundleIdentifierFromConfig === null) {
-    throw new Error(`Please define "ios.bundleIdentifier" in your ${configDescription}`);
-  } else {
-    if (bundleIdentifierFromPbxproj !== null) {
-      log(
-        `Using ${chalk.bold(
-          bundleIdentifierFromPbxproj
-        )} as the bundle identifier (read from the Xcode project).`
-      );
-      return bundleIdentifierFromPbxproj;
+  } else if (!bundleIdentifierFromPbxproj && !bundleIdentifierFromConfig) {
+    throw new Error(
+      `Please define "ios.bundleIdentifier" in your ${configDescription} and run "eas build:configure" again.`
+    );
+  } else if (bundleIdentifierFromPbxproj && !bundleIdentifierFromConfig) {
+    if (getConfigFilePaths(projectDir).staticConfigPath) {
+      await updateAppJsonConfigAsync(projectDir, exp, bundleIdentifierFromPbxproj);
     } else {
-      // bundleIdentifierFromConfig is never null in this case
-      // the following line is to satisfy TS
-      const bundleIdentifier = bundleIdentifierFromConfig ?? '';
-      log(
-        `Using ${chalk.bold(
-          bundleIdentifier
-        )} as the bundle identifier (read from ${configDescription}).
-We'll automatically configure your Xcode project using this value.`
+      throw new Error(
+        `Please define "ios.bundleIdentifier" in ${configDescription} and run "eas build:configure" again.`
       );
-      return bundleIdentifier;
     }
+  } else if (!bundleIdentifierFromPbxproj && bundleIdentifierFromConfig) {
+    IOSConfig.BundleIdenitifer.setBundleIdentifierForPbxproj(
+      projectDir,
+      bundleIdentifierFromConfig
+    );
+  }
+}
+
+async function updateAppJsonConfigAsync(
+  projectDir: string,
+  exp: ExpoConfig,
+  newBundleIdentifier: string
+): Promise<void> {
+  const paths = getConfigFilePaths(projectDir);
+  assert(paths.staticConfigPath, "can't update dynamic configs");
+
+  const rawStaticConfig = await fs.readJSON(paths.staticConfigPath);
+  rawStaticConfig.expo = {
+    ...rawStaticConfig.expo,
+    ios: { ...rawStaticConfig.expo.ios, bundleIdentifier: newBundleIdentifier },
+  };
+  await fs.writeJson(paths.staticConfigPath, rawStaticConfig, { spaces: 2 });
+
+  exp.ios = { ...exp.ios, bundleIdentifier: newBundleIdentifier };
+}
+
+/**
+ * Check if static config exists and if ios.bundleIdentifier is defined there.
+ * It will return false if value in static confgi is different than exp.ios.bundleIdentifier
```

Comment:
```suggestion
 * It will return false if value in static config is different than exp.ios.bundleIdentifier
```

Reply:
Depending on what editor you have @wkozyra95 (I use VSCode), you can get a autocorrect linter, I use this one https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker